src/rust/engine/src/core.rs

```diff
@@ -14,36 +13,46 @@ use handles::Handle;
 pub type FNV = hash::BuildHasherDefault<FnvHasher>;
 
 ///
-/// Variants represent a string->string map. For hashability purposes, they're stored
-/// as sorted string tuples.
+/// Params represent a TypeId->Key map.
+///
+/// For efficiency and hashability, they're stored as sorted Keys (with distinct TypeIds), and
```

Comment:
This description made me assume this would be a `Cow<BTreeSet<Key>>` rather than an `Arc<Vec<Key>>`...